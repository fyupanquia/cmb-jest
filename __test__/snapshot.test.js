import { getCharacter } from "../snapshot";
import rick from "../rick.json";
// > npx jest -u snapshot
describe("Es hora de las intantaneas", () => {
  test("Snapshot", () => {
    expect(getCharacter(rick)).toMatchSnapshot();
  });
  test("Siempre fallara la instantanea", () => {
    const user = {
      createAt: new Date(),
      id: Math.floor(Math.random() * 20),
    };
    expect(user).toMatchSnapshot();
  });

  test("Tenemos una excepción dentro del código", () => {
    const user = {
      id: Math.floor(Math.random() * 20),
      name: "cmb",
    };
    expect(user).toMatchSnapshot({
      id: expect.any(Number),
    });
  });
});
