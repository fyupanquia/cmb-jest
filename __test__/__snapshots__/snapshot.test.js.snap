// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las intantaneas Siempre fallara la instantanea 1`] = `
Object {
  "createAt": 2020-07-01T21:40:20.471Z,
  "id": 17,
}
`;

exports[`Es hora de las intantaneas Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanchez",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`Es hora de las intantaneas Tenemos una excepsion dentro del codigo 1`] = `
Object {
  "id": Any<Number>,
  "name": "cmb",
}
`;
